python/podio/generator_utils.py

```diff
@@ -70,9 +70,10 @@ def _is_fixed_width_type(type_name):
 class DataType:
   """Simple class to hold information about a datatype or component that is
   defined in the datamodel."""
-  def __init__(self, klass):
+  def __init__(self, klass, schema_version=None):
```

Comment:
That is unfortunately not the case. I kept it separate here as chained/stacked yaml files may have different versions and thus I wanted to avoid "globals".
The default "None" we should have neither since in the future we will require a version to be there. And we want to spot the cases where we forgot,

Reply:
I think at least for the generation part having a "global" is fine, because these are always just "global for the current EDM" that is being generated. I.e. extension models use their schema version here and there is no cross contamination from the upstream EDM.

Didn't think about catching non-defined schema versions more gracefully with the `None` default value. I think we can keep it. Maybe we should then also add a warning here to make it easier to spot from the output?

Reply:
I made it explicit now everywhere.